tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -106,34 +108,40 @@ public Span? RootSpan
                 spans.Array![spans.Offset].Context.TraceContext :
                 null;
 
-        internal void AddWafSecurityEvents(IReadOnlyCollection<object> events)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal AppSecRequestContext GetRequestContext()
         {
             if (Volatile.Read(ref _appSecRequestContext) is null)
             {
                 Interlocked.CompareExchange(ref _appSecRequestContext, new(), null);
             }
 
-            _appSecRequestContext!.AddWafSecurityEvents(events);
+            return _appSecRequestContext!;
```

Comment:
Wait, why do we need the `!` 🤔 Is it just different TFMs causing the issue, or is there a flow control issue. AFAICT, it _should_ be guaranteed non null, because `Interlocked.CompareExchange` ensures that?

Reply:
Yeah, without it it does not compile...